tensorflow_lite_support/ios/test/task/vision/image_searcher/TFLImageSearcherTests.swift

```diff
@@ -0,0 +1,178 @@
+    modelName: String
+  ) throws -> ImageSearcherOptions? {
+
+    let modelPath = try XCTUnwrap(filePath(name: modelName,
+        fileExtension: "tflite"))
+    return ImageSearcherOptions(modelPath: modelPath)
+  }
+
+  func createImageSearcher(
+    modelName: String,
+    indexFileName: String? = nil
+  ) throws -> ImageSearcher? {
+    let options = try XCTUnwrap(
+      self.createImageSearcherOptions(
+        modelName: "mobilenet_v3_small_100_224_searcher"))
+    
+
+    if let _indexFileName = indexFileName {
+      let indexFilePath = try XCTUnwrap(filePath(name: _indexFileName,
+          fileExtension: "ldb"))
+      options.searchOptions.indexFile.filePath = indexFilePath
+    }
+
+    let imageSearcher = try XCTUnwrap(
+      ImageSearcher.searcher(
+        options: options))
+
+    return imageSearcher
+  }
+ 
+
+  func validateSearchResult(
+    _ searchResult: SearchResult
+  ) {
+    self.validateSearchResultCount(
+      searchResult,
+      expectedNearestNeighborsCount: 5)
+    
+    self.validateNearestNeighbor(
+      searchResult.nearestNeighbors[0],
+      expectedMetadata: "burger",
+      expectedDistance: 198.456329)
+    self.validateNearestNeighbor(
+      searchResult.nearestNeighbors[1],
+      expectedMetadata: "car",
+      expectedDistance: 226.022186)
+    
+    self.validateNearestNeighbor(
+      searchResult.nearestNeighbors[2],
+      expectedMetadata: "bird",
+      expectedDistance: 227.297668)
+    self.validateNearestNeighbor(
+      searchResult.nearestNeighbors[3],
+      expectedMetadata: "dog",
+      expectedDistance: 229.133789)
+   self.validateNearestNeighbor(searchResult.nearestNeighbors[4],
+      expectedMetadata: "cat", 
+      expectedDistance: 229.718948)
+  }
+
+  func testSearchWithSearcherModelSucceeds() throws {
+    let imageSearcher = try XCTUnwrap(self.createImageSearcher(
+      modelName: kSearcherModelName))
+   
+    let mlImage = try XCTUnwrap(
+      MLImage.imageFromBundle(
+        class: type(of: self),
+        filename: "burger",
+        type: "jpg"))
+
+    let searchResult = try XCTUnwrap(
+      imageSearcher.search(
+        mlImage: mlImage))
+    self.validateSearchResult(searchResult)
+  }
+
+  func testSearchWithEmbedderModelAndIndexFileSucceeds() throws {
+    let imageSearcher = try XCTUnwrap(self.createImageSearcher(
+      modelName: kEmbedderModelName,
+      indexFileName: kMobileNetIndexName))
+   
+    let mlImage = try XCTUnwrap(
+      MLImage.imageFromBundle(
+        class: type(of: self),
+        filename: "burger",
+        type: "jpg"))
+
+    let searchResult = try XCTUnwrap(
+      imageSearcher.search(
+        mlImage: mlImage))
+    self.validateSearchResult(searchResult)
+  }
+
+  func testCreateImageSearcherWithNoModelPathFails() throws {
+    let options = ImageSearcherOptions()
+    do {
+      let imageSearcher = try ImageSearcher.searcher(
+        options: options)
+      XCTAssertNil(imageSearcher)
```

Comment:
An error is already thrown when you create the imageSearcher so this assert will not get called. You'll need to do the assert outside of the do-catch statement.

I'll fix it when I import the PR.